### ISSUE TITLE
Possible crash from hockeyApp Sdk

### ISSUE THREAD

**Issue by MidasCE** [bitstadium/HockeySDK-Android#429]
Hi, In our app we also have problem that may come from HockeyAppSDK, It would be great if you can help us.

```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.android.gms.internal.firebase-perf.na com.google.firebase.perf.internal.zzq.e()' on a null object reference
       at com.google.firebase.perf.internal.zzq.zza + 5(zzq.java:37)
       at com.google.android.gms.internal.firebase-perf.zzau.zzai + 3(zzau.java:112)
       at com.google.firebase.perf.network.zza.close + 10(zza.java:23)
       at java.io.BufferedInputStream.close + 485(BufferedInputStream.java:485)
       at net.hockeyapp.android.utils.Util.convertStreamToString + 43(Util.java:366)
       at net.hockeyapp.android.tasks.CheckUpdateTask.doInBackground + 11(CheckUpdateTask.java:115)
       at net.hockeyapp.android.tasks.CheckUpdateTask.doInBackground + 11(CheckUpdateTask.java:39)
       at android.os.AsyncTask$2.call + 333(AsyncTask.java:333)
       at java.util.concurrent.FutureTask.run + 266(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker + 1162(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run + 636(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run + 764(Thread.java:764)
```

This crash come after the HockeyApp run the update task.

I am wondering that the sdk may use some firebase services
(like push notifications, among a couple of things) that can cause this crash happen?

**Comment by annakocheshkova** [comment]
Hi! No, we don't use firebase in the SDK. 
However, the error implies that you don't have Google Services installed on your phone. Does this happen everytime and did it happen before on the same device?

**Comment by MidasCE** [comment]
It keep happen for users that use Android 7 and above. Here is my guess. 

UpdateManager call the CheckUpdateTask -> CheckUpdateTask is bound the app activity to HockeyWeb server -> Asynctask in hockeySdk got execute -> but when the input stream got close the activity or the firebase instance in the acitivity is not alive anymore -> crash happen.

**Comment by annakocheshkova** [comment]
Ah! Thanks @MidasCE I guess I see now exactly where and why it happens. We are closing the stream twice. Let me get back to you in a couple of hours.

**Comment by annakocheshkova** [comment]
So this is a confirmed bug. We will notify you once it is fixed, thank you.

**Comment by MidasCE** [comment]
Thanks @annakocheshkova

**Comment by MatkovIvan** [comment]
Fixed by #431
